src/widgets/workspace_bar.rs

```diff
@@ -72,48 +72,45 @@ impl<Msg> Widget<Msg, cosmic::Theme, cosmic::Renderer> for WorkspaceBar<'_, Msg>
         renderer: &cosmic::Renderer,
         limits: &layout::Limits,
     ) -> layout::Node {
-        // TODO configurable
-        let spacing = 8.0;
-
-        /*
-        layout::flex::resolve(
-            layout::flex::Axis::Vertical,
-            renderer,
-            &limits,
-            iced::Padding::ZERO,
-            0.0, // spacing
-            iced::Alignment::Start,
-            &self.children,
-            &mut tree.children,
-        )
-        */
-
         if self.children.is_empty() {
             return layout::Node::new(limits.min());
         }
 
-        let total_spacing = spacing * (self.children.len().saturating_sub(1)).max(0) as f32;
-        let max_main =
-            (self.axis.main(limits.max()) - total_spacing) / self.children().len() as f32;
+        // TODO configurable
+        let spacing = 8.0;
+
+        let total_spacing = spacing * (self.children.len() - 1) as f32;
+        let max_main = (self.axis.main(limits.max()) - total_spacing) / self.children.len() as f32;
         let max_cross = self.axis.cross(limits.max());
         let mut total_main = 0.0;
+        let mut available_cross = 0.0;
         let nodes = self
             .children
             .iter()
             .zip(tree.children.iter_mut())
-            .map(|(child, tree)| {
+            .enumerate()
+            .map(|(i, (child, tree))| {
                 let (max_width, max_height) = self.axis.pack(max_main, max_cross);
                 let child_limits =
                     layout::Limits::new(Size::ZERO, Size::new(max_width, max_height));
                 let mut layout = child.as_widget().layout(tree, renderer, &child_limits);
+                let child_size = layout.size();
                 let (x, y) = self.axis.pack(total_main, 0.0);
                 layout = layout.move_to(Point::new(x, y));
-                total_main += self.axis.main(layout.size()) + spacing;
+                available_cross = f32::max(available_cross, self.axis.cross(child_size));
+                let main = self.axis.main(child_size);
+                // XXX Don't add spacing for 0 length `dnd_source` placeholder widget
+                if main != 0.0 {
```

Comment:
Is there a reason this part was copied from https://github.com/pop-os/cosmic-workspaces-epoch/pull/128? Not really an issue, I suppose, but it seems it's not needed to add the `i < self.children.len() - 1` check.